common/src/com/thoughtworks/go/config/materials/AbstractMaterial.java

```diff
@@ -42,13 +39,15 @@
 
     protected CaseInsensitiveString name;
     protected String type;
+    private boolean requiresDestinationFolder;
     private Map<String, Object> sqlCriteria;
     private Map<String, Object> attributesForXml;
     private String pipelineUniqueFingerprint;
     private String fingerprint;
 
-    public AbstractMaterial(String typeName) {
+    public AbstractMaterial(String typeName, boolean requiresDestinationFolder) {
```

Comment:
Handle this as a class definition and not an instance definition.
For e.g. the scm or pluggable scm instance cannot be created with false as it will break behavior. 
This must be a protected method which can be overriden in the class definition.